code/web/src/modules/survey/Survey.js

```diff
@@ -47,31 +44,64 @@ class Survey extends PureComponent {
           this.props.previousPage(this.props.survey.page)
         }
 
+        genDesc = (selected) => {
+          const sorted = Object.keys(selected).sort((a, b) => selected[b] - selected[a])
+          if (sorted.length > 1) {
+            return `${styleToString(sorted[0])} and ${styleToString(sorted[1])}`
+          } else {
+            return styleToString(sorted[0])
+          }
+        }
```

Comment:
This is a very concise way to handle this data. It is a little confusing at first, but when you see that the array is being sorted by the value of it's key in the corresponding object, it is clear.